torch/csrc/distributed/rpc/ProcessGroupAgent.cpp

```diff
@@ -58,12 +58,15 @@ Message deserialize(std::istream& is) {
 ProcessGroupAgent::ProcessGroupAgent(
     std::string workerName,
     std::unordered_map<std::string, int> nameMap,
-    std::shared_ptr<c10d::ProcessGroup> pg)
+    std::shared_ptr<c10d::ProcessGroup> pg,
+    int numSendRecvThreads)
     : RpcAgent(std::move(workerName), processRequestBlocking),
       nameMap_(std::move(nameMap)),
       stop_(false),
       pg_(std::move(pg)),
-      nextId_(0) {
+      nextId_(0),
+      sendMutexes_(new std::mutex[pg_->getSize()]),
```

Comment:
Use `std::vector` instead?

Reply:
We need dynamic allocation here, and I thought we cannot resize vector of `mutex` because there is no copy/move constructor for `mutex`? Let me try. I also prefer to use `vector` if it is possible.

Reply:
You can use `vector` as long as you never resize it